augur/io.py

```diff
@@ -35,6 +36,66 @@ def open_file(path_or_buffer, mode="r", **kwargs):
         yield path_or_buffer
 
 
+def read_metadata(metadata_file, valid_index_cols=("strain", "name"), **kwargs):
+    """Read metadata from a given filename and into a pandas `DataFrame` or
+    `TextFileReader` object.
+
+    Parameters
+    ----------
+    metadata_file : str
+        Path to a metadata file to load.
+    valid_index_cols : list[str]
+        List of possible index column names to check for, ordered by priority.
+    kwargs : dict
+        Keyword arguments to pass through to pandas.read_csv.
+
+    Returns
+    -------
+    pandas.DataFrame or pandas.TextFileReader
+
+    Raises
+    ------
+    KeyError :
+        When the metadata file does not have any valid index columns.
+
+    """
+    default_kwargs = {
+        "sep": None,
+        "engine": "python",
+        "skipinitialspace": True,
+        "dtype": {
+            "strain": "string",
+            "name": "string,"
```

Comment:
Tom notes that these dtypes don't reflect the index columns provided by the argument above.

Reply:
Fixed in e975fef